tests/OpenTK.Tests/Generators.fs

```diff
@@ -22,18 +22,21 @@ module private Generators =
         singleArb
         |> Gen.two
         |> Gen.map Vector2
+        |> Gen.filter (fun v -> not <| (Single.IsNaN v.Length || Single.IsInfinity v.Length ))
```

Comment:
A simpler (and much faster) variant of this would be to create a new 'validSingleArb' which works like our single arb, but performs the filtering.

We would then not need to duplicate the filtering for each of the vector types.